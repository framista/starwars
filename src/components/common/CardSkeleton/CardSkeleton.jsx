import React from 'react';

export const CardSkeleton = () => (
  <div className="w-[500px] max-w-[90%] bg-white rounded shadow-2xl p-5">
    <div className="h-6 rounded-sm bg-gray-200 animate-pulse mb-4" />
    <div className="grid gap-1">
      <div className="w-[40%] h-4 rounded-sm bg-gray-200 animate-pulse" />
      <div className="w-[60%] h-4 rounded-sm bg-gray-200 animate-pulse" />
      <div className="w-[65%] h-4 rounded-sm bg-gray-200 animate-pulse" />
      <div className="w-[50%] h-4 rounded-sm bg-gray-200 animate-pulse" />
    </div>
  </div>
);

export default React.memo(CardSkeleton);
