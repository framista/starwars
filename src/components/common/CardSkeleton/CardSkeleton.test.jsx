import { render } from '@testing-library/react';

import { CardSkeleton } from './CardSkeleton';

describe('CardSkeleton', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<CardSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
