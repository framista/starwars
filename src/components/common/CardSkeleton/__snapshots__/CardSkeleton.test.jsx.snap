// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSkeleton should render correctly 1`] = `
<DocumentFragment>
  <div
    class="w-[500px] max-w-[90%] bg-white rounded shadow-2xl p-5"
  >
    <div
      class="h-6 rounded-sm bg-gray-200 animate-pulse mb-4"
    />
    <div
      class="grid gap-1"
    >
      <div
        class="w-[40%] h-4 rounded-sm bg-gray-200 animate-pulse"
      />
      <div
        class="w-[60%] h-4 rounded-sm bg-gray-200 animate-pulse"
      />
      <div
        class="w-[65%] h-4 rounded-sm bg-gray-200 animate-pulse"
      />
      <div
        class="w-[50%] h-4 rounded-sm bg-gray-200 animate-pulse"
      />
    </div>
  </div>
</DocumentFragment>
`;
